src/System.Management.Automation/engine/SessionStateContainer.cs

```diff
@@ -2382,16 +2382,38 @@ private bool HasGetChildItemDynamicParameters(ProviderInfo providerInfo)
                         // Since the path contained glob characters or we are recursing and the
```

Comment:
@iSazonov, your last commit had 17 failures in `PowerShell-CI-macos`
(These are 5 of the failures)

Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Recurse -Name.Get-ChildItem -Path -Recurse -Name
```powershell
Expected 8, but got 13.
at <ScriptBlock>, /Users/runner/runners/2.168.2/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/FileSystemProviderExtended.Tests.ps1: line 230
230:             $result.Count | Should -Be 8
```
Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Recurse -Name.Get-ChildItem -Path -Recurse -Name -Hidden
```powershell
Expected 4, but got 0.
at <ScriptBlock>, /Users/runner/runners/2.168.2/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/FileSystemProviderExtended.Tests.ps1: line 236
236:             $result.Count | Should -Be 4
```
Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Recurse -Name.Get-ChildItem -Path -Recurse -Name -Directory
```powershell
Expected 2, but got 3.
at <ScriptBlock>, /Users/runner/runners/2.168.2/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/FileSystemProviderExtended.Tests.ps1: line 256
256:             $result.Count | Should -Be 2
```
Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Recurse -Name.Get-ChildItem -Path -Recurse -Name -Attributes Directory
```powershell
Expected 2, but got 3.
at <ScriptBlock>, /Users/runner/runners/2.168.2/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/FileSystemProviderExtended.Tests.ps1: line 264
264:             $result.Count | Should -Be 2
```
Extended FileSystem Provider Tests for Get-ChildItem cmdlet.Validate Get-ChildItem -Path -Recurse -Name.Get-ChildItem -Path -Recurse -Name -File
```powershell
Expected 6, but got 10.
at <ScriptBlock>, /Users/runner/runners/2.168.2/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/FileSystemProviderExtended.Tests.ps1: line 272
272:             $result.Count | Should -Be 6
```

Reply:
Oh, I guess the tests fail because "TestDrive:" on MacOs is a symlink.

@vexx32 @mklement0 Could you please advise how to fix this in better way?

Reply:
> Oh, I guess the tests fail because "TestDrive:" on MacOs is a symlink.
> 
> @vexx32 @mklement0 Could you please advise how to fix this in better way?

I haven't looked through the code, but if you aren't already - try using `FileSystemEntry.OriginalRootDirectory` or `FileSystemEntry.ToSpecifiedPath()`.

Reply:
@SeeminglyScience Thanks!

Can anybody check this on Mac?